Ch 5 review (#485)
worldbank/dime-data-handbook: chapters/5-processing.tex

represents one observation.
This may seem trivial, but should not be taken for granted.
This section will also present what we call a \textit{tidy} data format,
which is, in our experience, the ideal format to handle tabular datasets.

\subsection{Unique identifiers}
% Uniquely and fully identifying variable
An important step before starting to tidy a dataset is
to understand the \textbf{unit of observation}\sidenote{
	\textbf{Unit of observation:} the unit described by the data. 
	In datasets, it is ideally what each row represents. \url{
		https://dimewiki.worldbank.org/wiki/Unit\_of\_Observation}}
and find out which variable or set of variables is the \textbf{unique identifier}\sidenote{\url{
		https://dimewiki.worldbank.org/ID\_Variable\_Properties}} for each observation in your dataset.
Ensuring that observations are uniquely and fully identified
is arguably the most important step in data cleaning.
It may be the case that the variables expected to uniquely identify 
the raw data contain either missing or duplicate values.\sidenote{
	Throughout this book, we use the expression \textbf{raw data}
	to refer to data in the state it was originally received by the research team.

> 	This applies to data acquired from partners as well as originally collected data. 

Reviewer: ```suggestion
	This applies to data acquired from partners as well as 
	original data  collected by the research team. 
```

> 	This applies to data acquired from partners as well as originally collected data. 

Reviewer: ```suggestion
	This applies to data acquired from partners as well as 
	original data  collected by the research team. 
```

Every other format is \textit{untidy}.
This may seem trivial, but raw data,
and raw survey data in particular,
is rarely received in a tidy format.
To contribute to the confusion,
in Stata each column is called a ``variable'',
and each row an ``observation'',
even though these do not necessarily equate.

Every untidy dataset is untidy in its own way,
but the most common case of untidy raw data encountered in development research
is a dataset with multiple units of observations stored in the same data table.
Take, for example, a household survey that includes household-level questions,
as well as a household member roster.
Such raw datasets usually consists of a single data table
where questions from the household member roster are saved in different columns,
one for each member, with a corresponding member suffix,
and household-level questions are represented by one column each.
When your rows include multiple nested observational units,
then the identifying variable does not identify all observations on that row,
as there is more than one unit of observation on the same row.
Survey data containing nested units of observation is typically

> imported from survey platforms in this format,
> which is called \textbf{wide format}.\sidenote{\textbf{Wide data:}

Author: ```suggestion
imported from survey platforms \textbf{wide format}.\sidenote{\textbf{Wide data:}
```

	a data table where a single variable is divided into multiple columns,
	for example one for each individual in a household.
}

> Wide data could have, for instance,

Author: ```suggestion
Wide format data could have, for instance,
```

> one column for the household variable called \texttt{ownsfridge},
> but a few columns for the household member variables \texttt{sex\_1}, \texttt{sex\_2}, etc.

Author: ```suggestion
one column for a household-level variable (e.g. \texttt{ownsfridge}),
and a few columns for household member-level variables (e.g. \texttt{sex\_1}, \texttt{sex\_2}).
```

Raw data is often saved in this format because it's the most efficient way to transfer it:
adding different levels of observation into the same data table
allows for data to be transferred in a single file.
However, this leads to the widespread practice of interacting with data in wide format,
although doing so is often inefficient and error-prone.

To understand how dealing with wide data can be complicated,
imagine you need to calculate the share of women
in each household using the household level data described above.
In a wide data table you will either have to first create variables counting
the number of women and the total number of household members,
and then calculate the share,
or you will have to transform the data to a different format.
In a tidy data table, however, where each row is a household member,
you can easily aggregate the share of women by household,
without additional steps,

> and then merge the result to the other data tables.

Author: ```suggestion
and then merge the result to the household-level data tables.
```

Tidy data tables are also easier to clean,
as each attribute only needs to be checked once,
and each column corresponds directly to one question in the questionnaire.
Finally, as you will see in Chapter 6,
summary statistics and distributions are much simpler
to generate from tidy data tables.

> Wide format is only one way in which data can be untidy.
> As mentioned earlier, there are unlimited ways for data to be untidy.

Author: ```suggestion
As mentioned earlier, there are unlimited ways for data to be untidy; wide format is only one of those ways.
```

first, identify all the variables that were measured at the same level of observation;
second, create separate data tables for each level of observation;
and third, reshape\sidenote{\textbf{Reshape:}
	transform a data table in such a way that the unit of observation represented by a row changes.}
the data and remove duplicated rows
until each data table is uniquely and fully identified by the identifying variable
that corresponds to its unit of observation.
Reshaping data tables is the most intricate task in data cleaning;
you should be very familiar with commands such as
\texttt{reshape} in Stata and \texttt{pivot} in R.
You must be sure that identifying variables are consistent across data tables,
so they can always be linked.
Reshaping is the type of transformation we referred to
in the example of how you calculate 
the share of women in a wide data set. 
The important difference is that 
in a tidy workflow,
instead of transforming the data for each operation,
this transformation is done once for all data during cleaning,
making all subsequent operations much easier.

In the earlier household survey example,
household-level variables will be stored in one tidy data table,

> and household-member variables are reshaped to member level
> and stored in another tidy data table,

Author: ```suggestion
and household-member variables are reshaped
and stored in a separate, member-level, tidy data table,
```

which also contains the household ID for each individual.
The household ID is intentionally duplicated in the household members data table
to allow one or several household members to be linked to the same household data.

> The household members data will be uniquely identified by
> either a single household member ID or
> a combination of household ID and household member ID.

Author: ```suggestion
The unique identifier for the household member-level data data will be
either a single household member ID or
a combination of household ID and household member ID.
```

In the transaction data example,
the result of the tidying process would be one transaction-level data table,
containing variables indicating the ID of all firms involved;
and one firm-level data table with a single entry for each firm.
Then, firm-level analysis is easily done
by calculating appropriate statistics in the transactions data table
(in Stata, often through \texttt{collapse})
and then merging or joining those results to the firms data table.

> In both examples above, your dataset is made up of two data tables,
> each saved in a separate file.

Author: ```suggestion
In both examples above, your clean dataset is made up of two tidy data tables,
each saved in a separate file.
```

> In both examples above, your dataset is made up of two data tables,
> each saved in a separate file.
> In a tidy workflow, your dataset is a set of one or more data tables,
> where each data table is a file in your data storage.

Author: ```suggestion
In a tidy workflow, your clean dataset is a set of one or more tidy data tables. In both examples above, your clean dataset is made up of two tidy data tables.

```

Author: I didn't understand what "where each data table is a file in your data storage" means. i removed but if you want to add back in should clarify

> There must also be a clear way to connect each

Author: ```suggestion
There must be a clear way to connect each
```

tidy data table to a master dataset,
and thereby also to all other datasets.

> This is done by deciding which data table is the main data table,

Author: ```suggestion
To implement this, you need to decide which data table is the main data table,
```

> and that data table's unit of observation is

Author: ```suggestion
that data table's unit of observation will be
```

the main unit of observation of your dataset.

> The main unit of observation must have a master dataset,

Author: must "have" a master dataset? i don't really know what that means. perhaps, must directly correspond to a master dataset?

Reviewer 2: change to correspond to

Reviewer 2: ```suggestion
The main unit of observation must correspond to a master dataset,
```

and be listed in the data linkage table.
All other data tables in your dataset must have
an unambiguous way to merge with the main data table.
This way, it will be possible to link
all data points in all your project's datasets to each other.
We recommend that you save your datasets as a folder,

> where the main data table share the same name as the folder,

Author: ```suggestion
in which the main data table shares the same name as the folder,
```

and the name of all other data tables start with the same name,
but are suffixed with the unit of observation for that data table.

> In the household dataset example, 

Author: i think it would be helpful to describe in this paragraph of how the folder and files would be named, at least for one of the examples.

Reviewer 2: I will do.

Reviewer 2: @mariaruth , done in 28d259b787e504fd437b396b9845fc363481d738, please review

> the household data table would be the main table.

Author: ```suggestion
the household-level data table would be the main table.
```

This means that there must be a master dataset for households.
You may have a master dataset for household members as well,
if you think it is important for your research,
but based on the information provided here,

> it is not required.

Author: Suggest removing this, or at least making it all parenthetical, since we don't really explain how one decides what master datasets are necessary

Reviewer: ```suggestion
(You may have a master dataset for household members as well,
if you think it is important for your research,
but it is not strictly required).
```

In the firm and transaction dataset example
it is not as obvious which is the main unit of observation.
It will depend on your research or how the data was collected.
Simply pick the one that makes cleaning your data the easiest,

> and make sure it is made clear which one you picked.

Author: I would just remove this, and focus on the hh example; this doesn't add much

Reviewer 2: I remove

Reviewer: ```suggestion
```

compose a consistent and useful grammar to perform the same tasks.

%--------------------------------------------------------------
\subsection{Documenting data cleaning}

Throughout the data cleaning process,
you will often need extensive inputs from the people responsible for data collection.
Sometimes this is your research team, but often it will be someone else.
It could be a survey team, the government ministry responsible for administrative data systems,
the technology firm that generated remote sensing data, etc.
Regardless of who originally collected the data,
you should acquire and organize all documentation of how the data was generated.\sidenote{
	\url{https://dimewiki.worldbank.org/Data\_Documentation}}
\index{Documentation}
What type of documentation that is available depends on how the data was collected.
For original data collection, this should include
field protocols, data collection manuals, survey instruments,
supervisor notes, and data quality monitoring reports.
For secondary data, you should try to get the same type of information,
but that is often not possible unless
the data source is a well managed data publication.

> The data documentation should be stored

Reviewer: ```suggestion
Independently of what exactly it consists of,
the data documentation should be stored
```
